### PR TITLE
feat: support import of Certificate and PrivateKey

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

**Review by ccorsin — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       subcommands: |
         python3 scripts/download_wasm.py
         npm install
-      kms-version: 4.5.0
+      kms-version: feature-pki
```
Comment (Manuthor) [inline .github/workflows/ci.yml:19]: as a memo, before merging this PR, all `feature-pki` must be replaced by `4.6.0` as soon the KMS v4.6.0 is released

File: scripts/download_wasm.py [hunk 2]
```diff
@@ -74,6 +74,6 @@ def download_wasm(version: str) -> bool:
 
 
 if __name__ == '__main__':
-    RET = download_wasm('v2.2.1')
+    RET = download_wasm('v2.2.2')
```
Comment (Manuthor) [inline scripts/download_wasm.py:77]:
```suggestion
    RET = download_wasm('v2.2.3')
```

File: src/kms/kms.ts [hunk 3]
```diff
@@ -316,6 +331,43 @@ export class KmsClient {
     )
   }
 
+  /**
+   * Import a X509 certificate as PEM encoded
```
Comment (Manuthor) [inline src/kms/kms.ts:335]:
```suggestion
   * Import a X509 certificate or a private key (both as PEM encoded)
```

File: src/kms/kms.ts [hunk 4]
```diff
@@ -316,6 +331,43 @@ export class KmsClient {
...
+   * @param {string} uniqueIdentifier  the unique identifier of the key
+   * @param {Uint8Array} certificateBytes the certificate as bytes
```
Comment (Manuthor) [inline src/kms/kms.ts:337]:
```suggestion
   * @param {Uint8Array} pemBytes the PEM certificate/private key as bytes
```

File: src/kms/kms.ts [hunk 5]
```diff
@@ -316,6 +331,43 @@ export class KmsClient {
...
+  public async importPem(
+    uniqueIdentifier: string,
+    certificateBytes: Uint8Array,
+    tags: string[] = [],
+    replaceExisting: boolean = false,
+  ): Promise<string> {
+    const attributes = new Attributes()
+    attributes.objectType = "Certificate"
+
+    const certificate = new Certificate(CertificateType.X509, certificateBytes)
```
Comment (Manuthor) [inline src/kms/kms.ts:351]:
```suggestion
    const pem = new Certificate(CertificateType.X509, pemBytes)
```

File: src/kms/kms.ts [hunk 6]
```diff
@@ -316,6 +331,43 @@ export class KmsClient {
...
+   * Import a X509 certificate as PEM encoded
+   * @param {string} uniqueIdentifier  the unique identifier of the key
+   * @param {Uint8Array} certificateBytes the certificate as bytes
+   * @param {string[]} tags potential list of tags
+   * @param {boolean} replaceExisting replace the existing object
+   * @returns {string}  the unique identifier of the key
+   */
+  public async importPem(
+    uniqueIdentifier: string,
+    certificateBytes: Uint8Array,
```
Comment (Manuthor) [inline src/kms/kms.ts:344]:
```suggestion
    pemBytes: Uint8Array,
```

File: src/kms/kms.ts [hunk 7]
```diff
@@ -316,6 +331,43 @@ export class KmsClient {
...
+        enc.encode(JSON.stringify(tags)),
+      )
+      attributes.vendorAttributes.push(vendor)
+    }
+
+    return await this.importObject(
+      uniqueIdentifier,
+      attributes,
+      attributes.objectType,
+      { type: "Certificate", value: certificate },
```
Comment (Manuthor) [inline src/kms/kms.ts:366]:
```suggestion
      { type: "Certificate", value: pem },
```

File: src/kms/kms.ts [hunk 8]
```diff
@@ -801,6 +856,65 @@ export class KmsClient {
...
     return publicMasterKey.policy()
   }
+
+  /**
+   * Get and wrap
+   * @param uniqueIdentifier the unique identifier of the object to get and wrap
+   * @param encryptionKeyUniqueIdentifier the unique identifier to use to wrap the fetched key
+   * @returns wrapped object
+   */
+  public async getWrappedKey(
```
Comment (Manuthor) [inline src/kms/kms.ts:866]: Do you think we could merge this function in the `getObject` function?
If we want to ease the use/creation of the `KeyWrappingSpecification` object, we could replace (in the actual `getObject` function) the `KeyWrappingSpecification` parameter by `encryptionKeyUniqueIdentifier` and let the `getObject` function recreate locally the `KeyWrappingSpecification`.

File: src/kms/kms.ts [hunk 9]
```diff
@@ -801,6 +856,65 @@ export class KmsClient {
...
+    uniqueIdentifier: string,
+    encryptionKeyUniqueIdentifier: string,
+  ): Promise<KmsObject> {
+    const wrappedObject = new KeyWrappingSpecification(
```
Comment (Manuthor) [inline src/kms/kms.ts:870]:
```suggestion
    const keyWrappingSpecification = new KeyWrappingSpecification(
```

File: src/kms/kms.ts [hunk 10]
```diff
@@ -801,6 +856,65 @@ export class KmsClient {
...
+      WrappingMethod.Encrypt,
+      new EncryptionKeyInformation(encryptionKeyUniqueIdentifier),
+    )
+    const object = await this.getObject(uniqueIdentifier, wrappedObject)
```
Comment (Manuthor) [inline src/kms/kms.ts:874]:
```suggestion
    const object = await this.getObject(uniqueIdentifier, keyWrappingSpecification)
```

File: tests/KMS.test.ts [hunk 11]
```diff
@@ -171,6 +175,30 @@ test(
...
+//       uuidv4(),
+//       new TextEncoder().encode(NIST_P256_PRIVATE_KEY),
+//       ["private key", "x509"],
+//     )
+//   },
+//   {
+//     timeout: 30 * 1000,
+//   },
+// )
+
```
Comment (Manuthor) [inline tests/KMS.test.ts:201]:
```suggestion

```

**Pull request merged**